packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,397 @@
+      basicAuth: {
+        type: "http",
+        scheme: "basic",
+      },
+      bearerAuth: {
+        type: "http",
+        scheme: "bearer",
+      },
+      ApiKeyAuth: {
+        type: "apiKey",
+        name: "api-key-header-name",
+        in: "header",
+      },
+    },
+  },
+})
+
+type PathGenerationErrors =
+  | "INVALID_METHOD"
+  | "INVALID_URL"
+  | "INVALID_CONTENT_TYPE"
+  | "INVALID_AUTH"
+  | RequestBodyGenerationErrors
+
+const generateOpenApiPathFromRequest = (
+  request: HoppRESTRequest
+): E.Either<
+  PathGenerationErrors,
+  OpenAPIV3.PathItemObject & { pathname: string }
+> =>
+  pipe(
+    E.Do,
+    E.bind("method", () =>
+      pipe(
+        request.method.toLowerCase(),
+        E.fromPredicate(isValidMethod, () => "INVALID_METHOD" as const)
+      )
+    ),
+    E.bindW("origin", () =>
+      pipe(
+        E.tryCatch(
+          () => new URL(request.endpoint),
+          () => "INVALID_URL" as const
+        )
+      )
+    ),
+
+    E.bindW("openApiBody", () =>
+      pipe(request.body, generateOpenApiRequestBody)
+    ),
+
+    E.bindW("openApiHeaders", () =>
+      pipe(request.headers, generateOpenApiHeaders, E.right)
+    ),
+
+    E.bindW("openapiQueryParams", () =>
+      pipe(request.params, generateOpenApiQueryParams, E.right)
+    ),
+
+    E.bindW("openapiAuth", () => pipe(request.auth, generateOpenApiAuth)),
+
+    E.map(
+      ({
+        method,
+        origin,
+        openApiBody,
+        openApiHeaders,
+        openapiQueryParams,
+        openapiAuth,
+      }): OpenAPIV3.PathItemObject & { pathname: string } =>
+        pipe({
+          pathname: origin.pathname,
+          [method]: {
+            description: request.name,
+            requestBody: openApiBody,
+            parameters: [...openApiHeaders, ...openapiQueryParams],
+            responses: {
+              200: {
+                description: "",
+              },
+            },
+            security: openapiAuth,
+          },
+          servers: [
+            {
+              url: origin.host,
+            },
+          ],
+        })
+    )
+  )
+
+const extractAllRequestsFromCollections = (
+  collections: HoppCollection<HoppRESTRequest>[]
+): HoppRESTRequest[] =>
+  pipe(
+    collections,
+    A.reduce([], (allRequests: HoppRESTRequest[], collection) => [
+      ...allRequests,
+      ...collection.requests,
+      ...extractAllRequestsFromCollections(collection.folders),
+    ])
+  )
+
+export const convertHoppToOpenApiCollection = (
+  collections: HoppCollection<HoppRESTRequest>[]
+): E.Either<string, OpenAPIV3.Document> =>
+  pipe(
+    collections,
+    extractAllRequestsFromCollections,
+    A.map(generateOpenApiPathFromRequest),
+    E.sequenceArray,
+    E.map(RA.toArray),
+    E.map(
+      flow(
+        A.reduce({}, (allPaths: OpenAPIV3.PathsObject, path) => ({
+          ...allPaths,
+          [path.pathname]: omit(path, "pathname"),
+        })),
+        generateOpenApiDocument
+      )
+    )
```

Comment:
Multiple maps can be put together into a single map.

```suggestion
    E.map(
      flow(
      	RA.toArray,
        A.reduce({}, (allPaths: OpenAPIV3.PathsObject, path) => ({
          ...allPaths,
          [path.pathname]: omit(path, "pathname"),
        })),
        generateOpenApiDocument
      )
    )
```